### PR TITLE
[BROWSEUI] CExplorerBand: Use SHCNRF_NewDelivery method

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Refactoring. Optimize for speed.
JIRA issue: [CORE-12619](https://jira.reactos.org/browse/CORE-12619)

## Proposed changes

- Update file headers.
- Make some parameter type `LPITEMIDLIST` `LPCITEMIDLIST` (Add `const`).
- Add `OnChangeNotify` method.
- Use `SHCNRF_NewDelivery` method for speed.
- Add some SAL annotations.

### CODE REVIEW

File: dll/win32/browseui/explorerband.cpp [hunk 1]
```diff
@@ -1,22 +1,9 @@
...
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ * PROJECT:     ReactOS Explorer
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
```
Comment (HBelusca) [inline dll/win32/browseui/explorerband.cpp:3]: Shouldn't that be "GPL" instead of LGPL?

**Comment by HBelusca** [comment]
> Use SHCNRF_NewDelivery method for speed.

But what is this so-called "new delivery" all about?

File: dll/win32/browseui/explorerband.cpp (hunk 1 above)
Comment (binarymaster) [reply dll/win32/browseui/explorerband.cpp:3]: image

File: dll/win32/browseui/explorerband.cpp [hunk 2]
```diff
@@ -1,22 +1,9 @@
...
+ * PURPOSE:     Explorer bar
+ * COPYRIGHT:   Copyright 2016 Sylvain Deverre <deverre dot sylv at gmail dot com>
```
Comment (binarymaster) [inline dll/win32/browseui/explorerband.cpp:5]:
```suggestion
 * COPYRIGHT:   Copyright 2016 Sylvain Deverre <deverre.sylv@gmail.com>
```

**Comment by katahiromz** [comment]
> > Use SHCNRF_NewDelivery method for speed.
> 
> But what is this so-called "new delivery" all about?

https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shchangenotifyregister


>SHCNRF_NewDelivery (0x8000)
Messages received use shared memory. Call [SHChangeNotification_Lock](https://learn.microsoft.com/en-us/windows/desktop/api/shlobj_core/nf-shlobj_core-shchangenotification_lock) to access the actual data. Call [SHChangeNotification_Unlock](https://learn.microsoft.com/en-us/windows/desktop/api/shlobj_core/nf-shlobj_core-shchangenotification_unlock) to release the memory when done.
>
>Note  We recommend this flag because it provides a more robust delivery method. All clients should specify this flag.

**Pull request merged**